runtime/lua/vim/lsp/handlers.lua

```diff
@@ -159,6 +161,28 @@ M['textDocument/codeLens'] = function(...)
 end
 
 
+--see: https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_references
+M['textDocument/references'] =function(_,result, ctx, config)
```

Comment:
I split this out into its own handler because we were taking advantage of lua to pass an invalid second argument, and this now is the offset_encoding, unlike with symbols to list.